File: ts/src/CommandLineCommands/PromptCommands/PromptCreateCommand.ts
import { PromptBaseCommand } from './PromptBaseCommand';
import { ConfigFileScope } from '../../Configuration/ConfigFileScope';
import { PromptFileHelpers } from '../../Helpers/PromptFileHelpers';
import { PromptDisplayHelpers } from '../../Helpers/PromptDisplayHelpers';
import { ConsoleHelpers } from '../../Helpers/ConsoleHelpers';

/**
 * Command to create a new prompt.
 */
export class PromptCreateCommand extends PromptBaseCommand {
  public promptName?: string;
  public promptText?: string;

  constructor() {
    super();
    this.scope = ConfigFileScope.Local; // Default to local scope
  }

  getCommandName(): string {
    return 'prompt create';
  }

  isEmpty(): boolean {
    return !this.promptName || !this.promptText;
  }

  async executeAsync(_interactive: boolean): Promise<number> {
    if (!this.promptName) {
      ConsoleHelpers.writeErrorLine('Error: Prompt name is required.');
      return 1;
    }
    
    if (!this.promptText) {
      ConsoleHelpers.writeErrorLine('Error: Prompt text is required.');
      return 1;
    }

    try {
      // Remove leading slash if someone added it (for consistency)
      let cleanName = this.promptName;
      if (cleanName.startsWith('/')) {
        cleanName = cleanName.substring(1);
      }
      
      // Check if the prompt already exists
      const existingPromptFile = PromptFileHelpers.findPromptFile(cleanName);
      if (existingPromptFile) {
        ConsoleHelpers.writeErrorLine(`Error: Prompt '${cleanName}' already exists.`);
        return 1;
      }
      
      // Save the prompt
      const fileName = PromptFileHelpers.savePrompt(cleanName, this.promptText, this.scope ?? ConfigFileScope.Local);
      PromptDisplayHelpers.displaySavedPromptFile(fileName);
      
      return 0;
    } catch (ex) {
      const error = ex as Error;
      ConsoleHelpers.writeErrorLine(`Error creating prompt: ${error.message}`);
      return 1;
    }
  }
}

File: src/cycod/CommandLineCommands/PromptCommands/PromptCreateCommand.cs
using System;
using System.Threading.Tasks;

/// <summary>
/// Command to create a new prompt.
/// </summary>
class PromptCreateCommand : PromptBaseCommand
{
    /// <summary>
    /// The name of the prompt to create.
    /// </summary>
    public string? PromptName { get; set; }
    
    /// <summary>
    /// The text content of the prompt.
    /// </summary>
    public string? PromptText { get; set; }

    /// <summary>
    /// Constructor initializes the base command.
    /// </summary>
    public PromptCreateCommand() : base()
    {
        Scope = ConfigFileScope.Local; // Default to local scope
    }

    /// <summary>
    /// Checks if the command is empty (i.e., no prompt name or text provided).
    /// </summary>
    /// <returns>True if empty, false otherwise.</returns>
    public override bool IsEmpty()
    {
        return string.IsNullOrWhiteSpace(PromptName) || string.IsNullOrWhiteSpace(PromptText);
    }

    /// <summary>
    /// Gets the command name.
    /// </summary>
    /// <returns>The command name.</returns>
    public override string GetCommandName()
    {
        return "prompt create";
    }

    /// <summary>
    /// Execute the create command.
    /// </summary>
    /// <param name="interactive">Whether the command is running in interactive mode.</param>
    /// <returns>Exit code, 0 for success, non-zero for failure.</returns>
    public override async Task<object> ExecuteAsync(bool interactive)
    {
        return await Task.Run(() => 
        {
            if (string.IsNullOrWhiteSpace(PromptName))
            {
                ConsoleHelpers.WriteErrorLine("Error: Prompt name is required.");
                return 1;
            }
            
            if (string.IsNullOrWhiteSpace(PromptText))
            {
                ConsoleHelpers.WriteErrorLine("Error: Prompt text is required.");
                return 1;
            }

            return ExecuteCreate(PromptName, PromptText, Scope ?? ConfigFileScope.Local);
        });
    }

    /// <summary>
    /// Execute the create command for the specified prompt name, text, and scope.
    /// </summary>
    /// <param name="promptName">The name of the prompt to create.</param>
    /// <param name="promptText">The text content of the prompt.</param>
    /// <param name="scope">The scope to create the prompt in.</param>
    /// <returns>Exit code, 0 for success, non-zero for failure.</returns>
    private int ExecuteCreate(string promptName, string promptText, ConfigFileScope scope)
    {
        try
        {
            // Remove leading slash if someone added it (for consistency)
            if (promptName.StartsWith('/'))
            {
                promptName = promptName.Substring(1);
            }
            
            // Check if the prompt already exists
            var existingPromptFile = PromptFileHelpers.FindPromptFile(promptName);
            if (existingPromptFile != null)
            {
                ConsoleHelpers.WriteErrorLine($"Error: Prompt '{promptName}' already exists.");
                return 1;
            }
            
            // Save the prompt
            var fileName = PromptFileHelpers.SavePrompt(promptName, promptText, scope);
            PromptDisplayHelpers.DisplaySavedPromptFile(fileName);
            
            return 0;
        }
        catch (Exception ex)
        {
            ConsoleHelpers.LogException(ex, "Error creating prompt");
            return 1;
        }
    }
}